Tests/SwiftSDKGeneratorTests/EndToEndTests.swift

```diff
@@ -125,66 +139,261 @@ final class EndToEndTests: XCTestCase {
+  }
+
+  func testAarch64FromContainer() async throws {
+    try skipSlow()
+    try await buildTestcases(logger, config: config.withArchitecture("aarch64").withDocker())
+  }
+
+  func testX86_64FromContainer() async throws {
+    try skipSlow()
+    try await buildTestcases(logger, config: config.withArchitecture("x86_64").withDocker())
+  }
+}
+
+final class Swift60_UbuntuEndToEndTests: XCTestCase {
+  let config = SDKConfiguration(
+    swiftVersion: "6.0.2",
+    linuxDistributionName: "ubuntu",
+    architecture: "aarch64",
+    withDocker: false
+  )
+
+  private let logger = Logger(label: "swift-sdk-generator")
+
+  func testAarch64Direct() async throws {
+    try skipSlow()
+    try await buildTestcases(logger, config: config.withArchitecture("aarch64"))
+  }
+
+  func testX86_64Direct() async throws {
+    try skipSlow()
+    try await buildTestcases(logger, config: config.withArchitecture("x86_64"))
+  }
+
+  func testAarch64FromContainer() async throws {
+    try skipSlow()
+    try await buildTestcases(logger, config: config.withArchitecture("aarch64").withDocker())
+  }
+
+  func testX86_64FromContainer() async throws {
+    try skipSlow()
+    try await buildTestcases(logger, config: config.withArchitecture("x86_64").withDocker())
+  }
+}
+
+final class Swift59_RHELEndToEndTests: XCTestCase {
+  let config = SDKConfiguration(
+    swiftVersion: "5.9.2",
+    linuxDistributionName: "rhel",
+    architecture: "aarch64",
+    withDocker: true  // RHEL-based SDKs can only be built from containers
+  )
+
+  private let logger = Logger(label: "swift-sdk-generator")
+
+  func testAarch64FromContainer() async throws {
+    try skipSlow()
+    try await buildTestcases(logger, config: config.withArchitecture("aarch64").withDocker())
+  }
+
+  func testX86_64FromContainer() async throws {
+    try skipSlow()
+    try await buildTestcases(logger, config: config.withArchitecture("x86_64").withDocker())
+  }
+}
+
+final class Swift510_RHELEndToEndTests: XCTestCase {
+  let config = SDKConfiguration(
+    swiftVersion: "5.10.1",
+    linuxDistributionName: "rhel",
+    architecture: "aarch64",
+    withDocker: true  // RHEL-based SDKs can only be built from containers
+  )
+
+  private let logger = Logger(label: "swift-sdk-generator")
+
+  func testAarch64FromContainer() async throws {
+    try skipSlow()
+    try await buildTestcases(logger, config: config.withArchitecture("aarch64").withDocker())
+  }
+
+  func testX86_64FromContainer() async throws {
+    try skipSlow()
+    try await buildTestcases(logger, config: config.withArchitecture("x86_64").withDocker())
+  }
+}
+
+final class Swift60_RHELEndToEndTests: XCTestCase {
+  let config = SDKConfiguration(
+    swiftVersion: "6.0.2",
+    linuxDistributionName: "rhel",
+    architecture: "aarch64",
+    withDocker: true  // RHEL-based SDKs can only be built from containers
+  )
+
+  private let logger = Logger(label: "swift-sdk-generator")
```

Comment:
Can this logger become a default argument so that we don't have to explicitly initialize it in every class?